android/app/src/main/java/org/fossasia/openevent/activities/SearchActivity.java

```diff
@@ -0,0 +1,205 @@
+
+        handleVisibility();
+
+        globalSearchAdapter = new GlobalSearchAdapter(results, this);
+        searchRecyclerView.setAdapter(globalSearchAdapter);
+        searchRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        if (savedInstanceState != null && savedInstanceState.getString(SEARCH) != null) {
+            searchText = savedInstanceState.getString(SEARCH);
+        }
+
+        if (Intent.ACTION_SEARCH.equals(getIntent().getAction())) {
+            handleIntent(getIntent());
+        }
+
+    }
+
+    @Override
+    protected int getLayoutResource() {
+        return R.layout.activity_search;
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_home, menu);
+        SearchManager searchManager =
+                (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        searchView = (SearchView) menu.findItem(R.id.action_search_home).getActionView();
+        searchView.setSearchableInfo(
+                searchManager.getSearchableInfo(getComponentName()));
+
+        searchView.setOnQueryTextListener(this);
+        if (searchText != null) {
+            searchView.setQuery(searchText, true);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        if(query!=null) {
+            searchQuery(query);
+            searchText = query;
+        }
+        else{
+            results.clear();
+            handleVisibility();
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        searchView.clearFocus();
+        return true;
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    public void handleIntent(Intent intent) {
+        final String query = intent.getStringExtra(SearchManager.QUERY);
+        searchQuery(query);
+    }
+
+    private void searchQuery(String constraint) {
+
+        results.clear();
+
+        if (!TextUtils.isEmpty(constraint)) {
+            String query = constraint.toLowerCase(Locale.getDefault());
+            String wildcardQuery = String.format("*%s*", query);
+            addResultsFromTracks(wildcardQuery);
+            addResultsFromSpeakers(wildcardQuery);
+            addResultsFromLocations(wildcardQuery);
+            handleVisibility();
+        }
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        if (searchView != null) {
+            outState.putString(SEARCH, searchText);
+        }
+        super.onSaveInstanceState(outState);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        OpenEventApp.getEventBus().unregister(this);
+
+        if (searchView != null) searchView.setOnQueryTextListener(null);
```

Comment:
@iamareebjamal If this is how my `onChangeListener` is defined
```filteredTracks.addChangeListener(tracks -> {
            results.add("Tracks");
            results.addAll(tracks);
            globalSearchAdapter.notifyDataSetChanged();
            Timber.d("Filtering done total results %d", tracks.size());
            handleVisibility();
        });
```
Then how should I detach it? If it is anonymous it's instance would only exist withing the function scope right?